src/main/java/gov/nasa/jpl/mbee/mdk/generator/DocumentGenerator.java

```diff
@@ -277,11 +277,20 @@ public Section parseView(Element view) {
                     //if (expose.size() == 1 && expose.get(0) instanceof Diagram) {
                     for (Element ex : elementImports) {
                         if (ex instanceof Diagram) {
-                            Image image = new Image();
-                            List<Object> images = new ArrayList<Object>();
-                            images.add(ex);
-                            image.setTargets(images);
-                            viewSection.addElement(image);
+                            String diagramType = Application.getInstance().getProject().getDiagram((Diagram) ex).getDiagramType().getType();
```

Comment:
If the project is needed in an action, then it should be passed in in the constructor. This makes actions re-usable without being dependent on the current state of the application, e.g. which open project is currently in focus. No change necessary this time, but let's try to migrate to that pattern over time.